tests/test_demes2fwdpy11.py

```diff
@@ -521,3 +522,211 @@ def test_pulse_migration_matrix(self):
+        )
+        self.assertEqual(len(self.demog.model.set_migration_rates), 2)
+        for set_mig in self.demog.model.set_migration_rates:
+            self.assertTrue(set_mig.deme in [0, 2])
+            if set_mig.deme == 0:
+                self.assertTrue(np.all(set_mig.migrates == np.array([0.8, 0.1, 0.1])))
+            elif set_mig.deme == 2:
+                self.assertTrue(np.all(set_mig.migrates == np.array([0.1, 0.1, 0.8])))
+
+
+@check_valid_demography
+class TestContinuousMigrationsSplits(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.b = demes.Builder(
+            description="splits with migration", time_units="generations"
+        )
+        self.b.add_deme(name="A", epochs=[dict(start_size=100, end_time=100)])
+        self.b.add_deme(
+            name="B", ancestors=["A"], epochs=[dict(start_size=100, end_time=0)]
+        )
+        self.b.add_deme(
+            name="C", ancestors=["A"], epochs=[dict(start_size=100, end_time=50)]
+        )
+        self.b.add_deme(
+            name="D", ancestors=["C"], epochs=[dict(start_size=100, end_time=0)]
+        )
+        self.b.add_deme(
+            name="E", ancestors=["C"], epochs=[dict(start_size=100, end_time=0)]
+        )
+        self.b.add_migration(demes=["B", "C"], rate=0.1)
+        self.b.add_migration(demes=["B", "D"], rate=0.1)
+        self.b.add_migration(demes=["B", "E"], rate=0.1)
+        self.b.add_migration(demes=["D", "E"], rate=0.1)
+        self.g = self.b.resolve()
+        self.demog = fwdpy11.discrete_demography.from_demes(self.g, 1)
+
+    def test_migration_matrix(self):
+        M = np.zeros(25).reshape(5, 5)
+        M[0, 0] = 1
+        self.assertTrue(np.all(self.demog.model.migmatrix.M == M))
+        num_splits = 2
+        len_set_migs = 3 * num_splits + 2 + 3
+        self.assertEqual(len(self.demog.model.set_migration_rates), len_set_migs)
+
+
+@check_valid_demography
+class TestMigrationSums(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        yaml_string = """
+description: test model
+time_units: generations
+demes:
+  - name: A
+    epochs:
+      - end_time: 200
+        start_size: 100
+  - name: B
+    ancestors: [A]
+    epochs:
+      - end_time: 100
+        start_size: 100
+  - name: C
+    ancestors: [A]
+    epochs:
+    - end_time: 20
+      start_size: 100
+  - name: D
+    ancestors: [B]
+    epochs:
+      - end_time: 2
+        start_size: 100
+      - end_time: 0
+        start_size: 10
+  - name: E
+    ancestors: [B]
+    epochs:
+      - end_time: 2
+        start_size: 100
+      - end_time: 0
+        start_size: 30
+migrations:
+  - demes: [C, D]
+    rate: 6.228e-5
+  - demes: [D, E]
+    rate: 4.14e-5
+"""
+        temp_fname = "temp_yaml.yml"
+        with open(temp_fname, "w+") as f:
+            f.write(yaml_string)
+        self.demog = fwdpy11.discrete_demography.from_demes(temp_fname, 1)
+        os.remove(temp_fname)
+
+
+@check_valid_demography
```

Comment:
See examples in tests/conftest.py for examples of just returning the demes.Graph from a string literal.  Ideally, there'd be one function for each of the two YAMLs involved.  Do *not* mark them as `pytest.fixture`.

Then, the two functions can be used in a test as follows:

```py
def x():
    return 1

def y():
    return 2

@pytest.mark.parametrize("data", [x(), y()])
def test_it(data):
    assert data == 1 or data == 2, f"{data}"
```

If there's a cleaner way to do that using `pytest.fixture`, I'll report back.  But it will be nice to skip using temp files.